File: src/test/java/kitchenpos/products/tobe/fakes/FakeProductRepository.java
package kitchenpos.products.tobe.fakes;

import kitchenpos.products.tobe.domain.Product;
import kitchenpos.products.tobe.domain.ProductRepository;

import java.util.*;

public class FakeProductRepository implements ProductRepository {
    private final Map<UUID, Product> products = new HashMap<>();

    @Override
    public Product save(Product product) {
        products.put(product.getId(), product);
        return product;
    }

    @Override
    public Optional<Product> findById(UUID id) {
        return Optional.ofNullable(products.get(id));
    }

    @Override
    public List<Product> findAll() {
        return new ArrayList<>(products.values());
    }

    @Override
    public List<Product> findAllByIdIn(List<UUID> ids) {
        return null;
    }
}

File: src/test/java/kitchenpos/products/tobe/domain/FakeProductRepository.kt
package kitchenpos.products.tobe.domain

import jakarta.transaction.NotSupportedException
import kitchenpos.products.domain.Product
import kitchenpos.products.domain.ProductRepository
import java.util.*

class FakeProductRepository : ProductRepository {
    private val repository: MutableMap<UUID, Product> = mutableMapOf()

    override fun save(product: Product): Product {
        val product = product.apply {
            id = UUID.randomUUID()
        }
        repository[product.id] = product
        return product
    }

    override fun findById(id: UUID): Optional<Product> {
        return Optional.of(repository[id] ?: run {
            throw NoSuchElementException()
        })
    }

    override fun findAll(): MutableList<Product> {
        throw NotSupportedException()
    }

    override fun findAllByIdIn(ids: MutableList<UUID>): MutableList<Product> {
        throw NotSupportedException()
    }

    fun deleteAll() {
        repository.clear()
    }
}